contrib/intel/jenkins/summary.py

```diff
@@ -7,389 +8,493 @@
+        super().__init__(log_dir, prov, build_mode,
+                         f'SHMEM_{prov}_shmem_{build_mode}',
+                         f"shmem {prov} {build_mode}")
+        if self.prov == 'uh':
+            self.keyphrase = 'summary'
+            # Failed
+        if self.prov == 'isx':
+            self.keyphrase = 'scaling'
+            # Failed
+        if self.prov == 'prk':
+            self.keyphrase = 'solution'
+            # ERROR:
+        self.name = 'no_test'
+
+    def check_uh(self):
+        # (test_002) Running test_shmem_atomics.x: Test all atomics... OK
+        # (test_003) Running test_shmem_barrier.x: Tests barrier ... Failed
+        if "running test_" in self.line:
+            tokens = self.line.split()
+            for token in tokens:
+                if 'test_' in token:
+                    self.name = token
+            if tokens[len(tokens) - 1] == 'ok':
+                self.passes += 1
+            else:
+                self.fails += 1
+                self.failed_tests.append(self.name)
+        # Summary
+        # x/z Passed.
+        # y/z Failed.
+        if self.keyphrase in self.line: #double check
+            passed = self.log.readline().lower()
+            failed = self.log.readline().lower()
+            if self.passes != int(passed.split()[1].split('/')[0]):
+                print(f"passes {self.passes} do not match log reported passes " \
+                        f"{int(passed.split()[1].split('/')[0])}")
+            if self.fails != int(failed.split()[1].split('/')[0]):
+                print(f"fails {self.fails} does not match log fails " \
+                        f"{int(failed.split()[1].split('/')[0])}")
+
+    def check_prk(self):
+        if self.keyphrase in self.line:
+            self.passes += 1
+        if 'error:' in self.line or "exiting with" in self.line:
+            self.fails += 1
+            self.failed_tests.append(f"{self.prov} {self.passes + self.fails}")
+        if 'test(s)' in self.line:
+            if int(self.line.split()[0]) != self.fails:
+                print(f"fails {self.fails} does not match log reported fails " \
+                    f"{int(self.line.split()[0])}")
+
+    def check_isx(self):
+        if self.keyphrase in self.line:
+            self.passes += 1
+        if ('failed' in self.line and 'test(s)' not in self.line) or \
+            "exiting with" in self.line:
+            self.fails += 1
+            self.failed_tests.append(f"{self.prov} {self.passes + self.fails}")
+        if 'test(s)' in self.line:
+            if int(self.line.split()[0]) != self.fails:
+                print(f"fails {self.fails} does not match log reported fails " \
+                        f"{int(self.line.split()[0])}")
+
+    def check_fails(self):
+        if "exiting with" in self.line:
+            self.fails += 1
+            self.failed_tests.append(f"{self.prov} {self.passes + self.fails}")
+
+    def check_line(self):
```

Comment:
my comment was lost somehow, but the switch-like syntax works for callables, too (I believe it could even be a lambda if you want to define the callable in-place)

```
check = {
    "uh": self.check_uh,
    "isx": self.check_isx,
    "prk": self.check_prk
}
check[self.prov]()
```